### DISCUSSION TITLE
Should extsearch plugin be removed

### DISCUSSION THREAD

**Discussion by jevenski** [Novik/ruTorrent#2838] [General]
This plugin was designed many years ago to provide search for torrents from various trackers. Most trackers have evolved over the years but the plugin has been less actively maintained, so as a result we have higher chance seeing the search engine (a scraper script for searching an individual tracker) ending up in failures. I have tried one or two search engines and both of them were not working. We don't have enough maintainers to keep those up-to-date.

With the prosperity of *arr suites, however, we now have very powerful specialized toolsets for achieving the same goal. For example prowlarr for searching multiple trackers at the same time. The scrapers for *arr are more actively maintained by skilled programmers. Some of them might even be supported by the tracker officially. They can totally replace this extsearch plugin and achieve much more. I think it's time that we drop support for this plugin, and another two dependency plugins as well (cookies and login manager).

**Comment by koblack** [comment]
I agree that it can be removed. 

It would be great to replace it with a Jackett/Prowlarr integration. 
Someone already tried to add support for them in pull request #1753

**Comment by stickz** [comment]
I agree about removing it. We can't do it without anther major version bump though. We'll have to set a milestone for `v6`. Some people are still using these features. The plan is to offer long term support for `v5` throughout 2025. That would throw off plans to get everyone off `v4`.

One thing we can remove in `v5.2` is the **Dark** theme in favour of the **DarkBetter** theme. We're ready for the theme drop right now.

Development cycle wise, if we continue on our current mobile path, we're going to have a lot of success in increasing the ruTorrent userbase. rTorrent development just restarted a few months ago, which is really helpful. If we dedicate January 2025 to mobile support, I'll think we'll be able to start `v6` in February 2025. Worse case scenario, we're looking at March 2025, due to our limited maintainers.

**Comment by jevenski** [comment]
So shall we add some kind of deprecation warning or notice now to suggest users who are using this plugin to migrate to prowlarr or jackett (I haven't tried jackett yet)?

**Reply by stickz** [reply]
Nope, we can drop these plugins in version 6 without any deprecation warnings.
